File: web/src/services/category.service.ts
import { api } from './api';
import { Category } from '@/types';

interface CategoriesResponse {
  categories: Category[];
}

interface CategoryResponse {
  category: Category;
  message?: string;
}

export const categoryService = {
  async getCategories(): Promise<Category[]> {
    const response = await api.get<CategoriesResponse>('/categories');
    return response.categories;
  },

  async createCategory(data: { name: string; color?: string; icon?: string }): Promise<Category> {
    const response = await api.post<CategoryResponse>('/categories', data);
    return response.category;
  },

  async updateCategory(id: string, data: { name?: string; color?: string; icon?: string }): Promise<Category> {
    const response = await api.put<CategoryResponse>(`/categories/${id}`, data);
    return response.category;
  },

  async deleteCategory(id: string): Promise<void> {
    await api.delete(`/categories/${id}`);
  },
};

File: ios/RecipeApp/Services/CategoryService.swift
import Foundation

class CategoryService {
    static let shared = CategoryService()
    private let client = APIClient.shared

    private init() {}

    func getCategories() async throws -> [Category] {
        let response: CategoriesResponse = try await client.request(
            endpoint: "/categories",
            method: "GET"
        )
        return response.categories
    }

    func createCategory(_ request: CategoryRequest) async throws -> Category {
        let response: CategoryResponse = try await client.request(
            endpoint: "/categories",
            method: "POST",
            body: request
        )
        return response.category
    }

    func updateCategory(id: UUID, request: CategoryRequest) async throws -> Category {
        let response: CategoryResponse = try await client.request(
            endpoint: "/categories/\(id.uuidString)",
            method: "PUT",
            body: request
        )
        return response.category
    }

    func deleteCategory(id: UUID) async throws {
        let _: EmptyResponse = try await client.request(
            endpoint: "/categories/\(id.uuidString)",
            method: "DELETE"
        )
    }

    func addRecipeToCategory(recipeId: UUID, categoryId: UUID) async throws {
        let _: EmptyResponse = try await client.request(
            endpoint: "/categories/\(categoryId.uuidString)/recipes/\(recipeId.uuidString)",
            method: "POST"
        )
    }

    func removeRecipeFromCategory(recipeId: UUID, categoryId: UUID) async throws {
        let _: EmptyResponse = try await client.request(
            endpoint: "/categories/\(categoryId.uuidString)/recipes/\(recipeId.uuidString)",
            method: "DELETE"
        )
    }
}
